### PR TITLE
added github action to run tests to replace travis

### PR DESCRIPTION
**Pull request by mraarif**
**Description:**
Adds a GitHub action to run python tests

**JIRA:**
[BOM-2095](https://openedx.atlassian.net/browse/BOM-2095)

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (jmbowman) [inline .github/workflows/ci.yml:14]: Any particular reason for imposing this limit?  (Not that we'd hit it with the current configuration.)

File: .github/workflows/ci.yml
Comment (jmbowman) [inline .github/workflows/ci.yml:31]: Nit: missing newline

File: .github/workflows/ci.yml
Comment (jmbowman) [inline .github/workflows/ci.yml:30]: This will run the Python tests, and then run the quality checks in the same worker.  For this repo it's probably fast enough, but I think we should get in the habit of running these in parallel in different `job`s.  Every little speedup in time to getting feedback makes it less likely that developers will need to context switch while waiting for CI results.

**Review by jmbowman — COMMENTED** [review]
Looks good, just have a few suggestions for further improvement.  I want to make sure we've thought out how to best do this before we start copying this pattern across dozens of repositories.

File: requirements/pip.txt
Comment (jmbowman) [inline requirements/pip.txt:9]: Hmm, it looks like `--allow-unsafe` didn't work.  It seems to be working for other people, not sure if we're just using it wrong somehow or it broke recently.  Haven't had time to investigate in depth yet.

Comment (mraarif) [reply requirements/pip.txt:9]: This comment was added by pip-tools when I ran it without '--allow-unsafe' flag, pip and setuptools were commented out
It has now added the dependencies but didn't remove the comment

Comment (jmbowman) [reply requirements/pip.txt:9]: Ah, missed that the dependencies themselves aren't actually commented out.  Odd, but I guess that works.

File: .github/workflows/ci.yml
Comment (jmbowman) [inline .github/workflows/ci.yml:55]: A thought on reducing some of the duplication here: could we have just one job (maybe "run_tox") with a matrix entry like `toxenv: [py, quality]`?  Then we could use an `env` section in the job with `TOXENV: ${{ matrix.toxenv }}`.  Alternatively, we could use [tox-gh-actions](https://pypi.org/project/tox-gh-actions/) to do something very similar; it would add a dependency, but seems like it would also handle tox configs for testing multiple Django/celery/DRF versions more easily.

Comment (jmbowman) [reply .github/workflows/ci.yml:55]: Ah, I forgot that one of the drawbacks of `tox-travis` and `tox-gh-actions` is that they only utilize one worker per Python version, rather than one per tox env.  Probably more efficient to stick with the `TOXENV` approach.

Comment (mraarif) [reply .github/workflows/ci.yml:55]: I tried using (defaults)[https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#defaults] for GitHub workflows but it has its limitations

**Review by jmbowman — APPROVED** [review]
Thanks, that looks about as concise as we can get it while still being obvious what's happening.  I still think we should generally put the OS version into the matrix as per https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-running-with-more-than-one-operating-system (to facilitate future upgrades), but that can potentially wait until later.

Are you planning a followup PR to remove `.travis.yml` and maybe rename `requirements/travis.*`?

**Comment by mraarif** [comment]
looks like by removing `.travis.yml` Travis CI check won't complete and won't allow us to merge.

**Review by jmbowman — APPROVED** [review]
Looks good; once this is merged I'll update the branch protection rules to require passing GH Actions tests for merging instead of Travis.

**Comment by mraarif** [comment]
@jmbowman 
I noticed cookiecutter templates are still using `.travis.yml`, should we update these to generate apps with GitHub actions for CI instead?

**Comment by jmbowman** [comment]
Yes, I had that ticketed as part of ARCHBOM-1623 ; just moved it to [BOM-2131](https://openedx.atlassian.net/browse/BOM-2131) and moved it to the board.

**Comment by jmbowman** [comment]
Also, I just updated the branch protection rule on this repo.  Looking at the list of checks, I really like that our new configuration shows right in the commit's CI results the outcomes for each OS, Python version, and tox environment.  With Travis, you had to go over to their UI to tell anything beyond "CI passed" or "CI failed" (and you had to hunt through the logs to figure out which OS version was in use).

**Pull request merged**
